src/mongo_logger.ts

```diff
@@ -728,6 +728,7 @@ export class MongoLogger {
   logDestination: MongoDBLogWritable;
   logDestinationIsStdErr: boolean;
   pendingLog: PromiseLike<unknown> | unknown = null;
+  willLog: Record<MongoLoggableComponent, Record<SeverityLevel, boolean>>;
```

Comment:
```suggestion
  private severities: Record<MongoLoggableComponent, Record<SeverityLevel, boolean>>;
```

From a design perspective I don't think we should require other objects asking the logger to know about it's internal structure and this should be encapsulated behind an actual `willLog` method like before, since it was previously a public method and is now a publicly accessible attribute (even though it's called only internally for now). See other notes and suggestions on how I'd refactor this.

Reply:
I'm conflicted here, since we are using the willLog as a precomputed object in order to avoid the perf cost of calling a function every-time (access to a variable is quicker). How will this refactor affect perf?

Reply:
Additionally, I'm trying to think of a more precise name for `severities`, since it could get mixed up with `componentSeverities`.  In either case, I'll add a clarifying comment to both variables specifying their function.

Reply:
We would still be using a precomputed object to avoid the perf costs - you can see from my suggestions that still happens. The only difference here is that we access that through a method in my suggestions instead of accessing the precomputed object directly. This encapsulation of that access should not have an impact but also allows us to be able to refactor the underlying structure at any point in time without the need to refactor potential multiple calls to it in the future, it all goes through one method.

Reply:
I'm also completely fine using a different name than `severities` - I just called it that for the purposes of trying to explain in code what I was suggesting.

Reply:
I think since it's not a public facing variable, we don't have to be as careful with naming. The code comments should suffice imo.